packages/upgrade/src/actions/install.ts

```diff
@@ -0,0 +1,125 @@
+import type { Context, PackageInfo } from './context.js';
+
+import fs from 'node:fs';
+import path from 'node:path';
+import { fileURLToPath } from 'node:url';
+import { color, say } from '@astrojs/cli-kit';
+import { pluralize, celebrations, done, error, info, log, spinner, success, upgrade, banner, title, changelog, warn, bye } from '../messages.js';
+import { shell } from '../shell.js';
+import { random, sleep } from '@astrojs/cli-kit/utils';
+
+export async function install(
+	ctx: Pick<Context, 'version' | 'packages' | 'packageManager' | 'prompt' | 'dryRun' | 'exit' | 'cwd'>
+) {
+	await banner();
+	log('')
```

Comment:
nit: I would create a function called `printHardLine()`, so we know that these lone `log("")` are intentional

Reply:
Good feedback! Addressed in [377a543](https://github.com/withastro/astro/pull/8525/commits/377a543b9f5b62ffb08045b37f95326e0fbe535d)